src/config.h

```diff
@@ -15,28 +15,49 @@
 #include "util/config_macros.h"
 
 typedef enum {
-  FailurePolicy_Return,       // Return what we have on failure (timeout or OOM)
-  FailurePolicy_Fail,         // Just fail without returning anything
-  FailurePolicy_Invalid       // Not a real value
-} RSFailurePolicy;
-
-static const int on_failure_enums[2] = {
-  FailurePolicy_Return,
-  FailurePolicy_Fail
+  TimeoutPolicy_Return,       // Return what we have on timeout
+  TimeoutPolicy_Fail,         // Just fail without returning anything
+  TimeoutPolicy_Invalid       // Not a real value
+} RSTimeoutPolicy;
+
+static const int on_timeout_enums[2] = {
+  TimeoutPolicy_Return,
+  TimeoutPolicy_Fail
 };
-static const char *on_failure_vals[2] = {
+static const char *on_timeout_vals[2] = {
   "return",
   "fail"
 };
 
+typedef enum {
+  OomPolicy_Return,       // Return what we have on OOM
+  OomPolicy_Fail,         // Just fail without returning anything
+  OomPolicy_Ignore,       // Ignore OOM and continue
+  OomPolicy_Invalid       // Not a real value
```

Comment:
Why do we need the `Invalid` value? (regarding both enums)

Reply:
No, up to you. I think it's OK, especially if it's aligned with past configs. I don't think it adds anything, and we don't store invalid configs, so it should be unreachable